### PR TITLE
ci: run blacken and lint sessions on generated samples

### LINKED ISSUE #1191: Lint check fails for generated samples
The lint check is failing on python-bigquery. See build log [here](https://source.cloud.google.com/results/invocations/1c93f902-3d54-4da3-b108-7380c11c099c/log) from https://github.com/googleapis/python-bigquery/pull/1129

I can reproduce the issue with the showcase code:
```
partheniou@partheniou:~/git/gapic-generator-python/test$ nox -s lint
nox > Running session lint
nox > Creating virtual environment (virtualenv) using python3.9 in .nox/lint
nox > python -m pip install flake8 black==19.10b0
nox > black --check docs google tests samples noxfile.py setup.py
All done! ✨ 🍰 ✨
75 files would be left unchanged.
nox > flake8 google tests samples
samples/generated_samples/localhost:7469_generated_showcase_v1beta1_echo_collect_sync_98a2221d.py:33:5: F841 local variable 'client' is assigned to but never used
samples/generated_samples/localhost:7469_generated_showcase_v1beta1_identity_delete_user_sync_7345a63e.py:39:5: F841 local variable 'response' is assigned to but never used
samples/generated_samples/localhost:7469_generated_showcase_v1beta1_echo_collect_sync_bb1f4448.py:33:5: F841 local variable 'client' is assigned to but never used
samples/generated_samples/localhost:7469_generated_showcase_v1beta1_identity_delete_user_async.py:39:5: F841 local variable 'response' is assigned to but never used
samples/generated_samples/localhost:7469_generated_showcase_v1beta1_echo_collect_async.py:33:5: F841 local variable 'client' is assigned to but never used
samples/generated_samples/localhost:7469_generated_showcase_v1beta1_identity_delete_user_sync_a45fe6ed.py:39:5: F841 local variable 'response' is assigned to but never used
nox > Command flake8 google tests samples failed with exit code 1
nox > Session lint failed.
```

### PR DESCRIPTION
**Pull request by parthea**
Closes googleapis/gapic-generator-python#1191

### CODE REVIEW

**Comment by parthea** [comment]
Will rebase once https://github.com/googleapis/gapic-generator-python/pull/1193 is merged

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -316,6 +316,24 @@ jobs:
           echo "The old one will disappear after 7 days."
       - name: Integration Tests
         run: bazel test tests/integration:asset tests/integration:credentials tests/integration:logging tests/integration:redis
+  goldens-lint:
```
Comment (busunkim96) [inline .github/workflows/tests.yaml:319]: Trying to install 3.9 and run nox in the integration check resulted in weird errors about GLIBC version that I had trouble resolving. The integration tests already check the goldens match the output from the generator, so this just runs the noxfiles as present in `tests/integration/goldens/...`

**Review by busunkim96 — COMMENTED** [review]
I took over this PR and made the following changes:

* Added a new `goldens-lint` check that runs lint over the golden files produced by integration. @parthea originally had the additional `nox` sessions in the integration check, but I had trouble getting pip to work properly in that container.
* Fix lint errors in generated snippets:
    * properly print response for client streaming request 
    * omit `response = ...` when the RPC doesn't return anything 
    * Add an empty line between `requests = [request]` and `def request_generator` for client streaming

**Comment by busunkim96** [comment]
@parthea @software-dov This is ready for review

**Review by parthea — COMMENTED** [review]
LGTM

**Review by software-dov — APPROVED** [review]
LGTM. No rush, but could you file a tracker to port these changes to the ads templates?

**Comment by busunkim96** [comment]
@software-dov That is a good callout, I have https://github.com/googleapis/gapic-generator-python/issues/1052 to track Ads support for snippets.

**Pull request merged**